Sources/MapboxDirections/Intersection.swift

```diff
@@ -267,7 +267,11 @@ extension Intersection: Codable {
             lanes = approachLanes.map { Lane(indications: $0) }
             for i in usableApproachLanes {
                 lanes![i].isValid = true
-                if lanes![i].indications.contains(usableLaneIndication){
+//                if lanes![i].indications.contains(usableLaneIndication){
+//                    lanes![i].validIndication = usableLaneIndication
+//                }
+//                let usable = LaneIndication(descriptions: [usableLaneIndication.rawValue])
+                if lanes![i].indications.contains(LaneIndication(descriptions: [usableLaneIndication.rawValue])!) {
```

Comment:
This is implicitly reinterpreting a maneuver direction description string as a lane indication description string. That is what we want, but we should make this step more explicit. Let’s create a convenience initializer on `LaneIndication` that takes a `ManeuverDirection`. It can switch over all the anticipated description strings to create the corresponding `LaneIndication`, or it can use the same `rawValue`-based approach here, but at least it would be better documented that way.

Reply:
A better way to compare without the convenience initializer is to compare the string descriptions of the `LaneIndication` to the `ManeuverDirection`'s raw value:
`lanes![i].indications.descriptions.contains(usableLaneIndication.rawValue)`. If this comparison is used, is the convenience initializer necessary?

Reply:
I think it would be preferable to make the conversion explicit in some fashion. Otherwise it would be more difficult to find this raw value–based conversion if the assumption ever breaks in the future.